src/lib/repl_service_ctx.cpp

```diff
@@ -64,6 +64,18 @@ repl_service_ctx::repl_service_ctx(nuraft::raft_server* server) : _server(server
 
 bool repl_service_ctx::is_raft_leader() const { return _server->is_leader(); }
 
+const std::string& repl_service_ctx::raft_leader_id() const {
+    // when adding member to raft,  the id recorded in raft is a hash
+    // of passed-in id (new_id in add_member()), the new_id was stored
+    // in endpoint field.
+    static std::string const empty;
+    if (!_server) return empty;
+    auto const leader = _server->get_leader();
+    if (leader == -1) return empty;
+    auto const& srv_config = _server->get_srv_config(leader);
+    return (srv_config) ? srv_config->get_endpoint() : empty;
+}
+
```

Comment:
Can you add a unit test case to cover this method?

Reply:
Can we just make it simply:
```
std::string_view ...::raft_leader_id() const {
   if (_server) {
      if (auto leader = _server->get_srv_config(_server->get_leader()); nullptr != leader)
         return leader->get_endpoint();
   }
   return std::string_view();
}
```

Reply:
string_view is very tricky here as it is only a reference here (actually I think string & is also not right).  It is fully depends on the whole chane of `_server->get_srv_config()->get_endpoint()` is all return by a reference to a stable object, not returning a temp object.

Reply:
Since the intent is to stash this in _every_ response (Reads and Writes) probably worth any optimization, not just a memory but potential allocation. If it were purely a ctrl path operation I'd say no. What do you think?

Reply:
yes you are right. Code changed.

Out of curious,  any compiling time check can be made to gate if code are referencing to a local variable/tmp object?

i. e, assuming `get_string()` is an external API, once it change from returning a reference to return a temp object, how we can detect that in our code (`get_string_view_1`)
```
string b("abc");

string & get_string() {
  return b;
}

string get_string_1() {
  string c = "abc";
  return c;
}
string_view get_string_view() {
  return string_view(get_string());
}

string_view get_string_view_1() {
  //returning address of local temp object.
  return string_view(get_string_1());
}
```

Reply:
So the first assumption I have is that if someone changes the API they do so correctly by bumping the MAJOR version which would require someone to _look_ at the difference before updating. Bad assumption probably.

In your scenario, if someone started returning a temporary from nuraft_mesg then there must be a reason right? As in; it's not safe to dereference whatever nuraft::raft_server returned in the `cluster_config` value after the stack unwinds. So a copy is required...not sure what we as the library consumer can do about that and if we _should_? If it's someone making a de-optimized "bug"...I guess that's on the nuraft_mesg PR reviewer to catch?

Reply:
no It is not in nuraft_mesg but the nuraft core(external library), i.e , if sometimes the `get_endpoint()` returns a string instead of string ref.

For external project like nuraft, the semantic versioning is a handicraft maintained by you (https://github.com/szmyd/conan-center-index/blob/master/recipes/nuraft/all/conandata.yml)  and curious how you can identify the API changes to bumping MAJOR/MINOR version?    I might easily over-looking them .

Reply:
```c++
string_view get_string_view_1() {
  //returning address of local temp object will fail to compile
  static_assert(std::is_reference<decltype(get_string_1())>::value));
  return string_view(get_string_1());
}
```
?